### PR TITLE
Fix for Windows CreateFile long path

### PR DESCRIPTION
**Pull request by allexzander**
This should fix https://github.com/nextcloud/desktop/issues/3006

### CODE REVIEW

**Review by FlexW — APPROVED** [review]
Looks good to me.

**Review by camilasan — APPROVED** [review]

**Comment by cscholz** [comment]
Will this fix already available in the nightly build for windows?

**Comment by allexzander** [comment]
> Will this fix already available in the nightly build for windows?

@cscholz
Crossing all fingers. If Windows build succeeds, then, yes, it will be there.

File: src/libsync/vfs/cfapi/cfapiwrapper.cpp [hunk 1]
```diff
@@ -370,17 +371,24 @@ bool OCC::CfApiWrapper::isSparseFile(const QString &path)
 
 OCC::CfApiWrapper::FileHandle OCC::CfApiWrapper::handleForPath(const QString &path)
 {
+    if (path.isEmpty()) {
```
Comment (er-vin) [inline src/libsync/vfs/cfapi/cfapiwrapper.cpp:374]: I'd add a `Q_ASSERT(!path.isEmpty());` before this if. This is clearly an error in the caller when that happens so better crash hard and catch it in debug builds.

**Pull request merged**